core/src/com/mygdx/game/models/TerrainClass.java

```diff
@@ -0,0 +1,72 @@
+package com.mygdx.game.models;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * A class that represents a tile in the game.
+ */
+public class TerrainClass implements Terrain {
+    private String type;
+    private int defense;
+    private TerrainState terrainState;
+
+    public TerrainClass(String type, int defense, TerrainState terrainState) {
+        this.type = type;
+        this.defense = defense;
+        this.terrainState = terrainState;
+    }
+
+    /**
+     * Returns the Terrain from the configuration file.
+     *
+     * @param filename the filename
+     * @return the Terrain from the configuration file
+     */
+    public static TerrainClass getTerrainFromConfig(String filename) {
```

Comment:
can you make override the clone method of this class

Reply:
'Object.clone() is not present in JRE Emulation Library.' With copy constructor?

Reply:
Mmh do what ever you think is best

Reply:
Switching to Android Studio fixed the JRE Emulation problem. Currently using a copy constructor to clone objects.